src/de/raw.rs

```diff
@@ -454,12 +458,16 @@ impl<'de, 'a> serde::de::Deserializer<'de> for &'a mut Deserializer<'de> {
 
                 self.deserialize_next(visitor, DeserializerHint::RawBson)
             }
+            HUMAN_READABLE_NEWTYPE => {
+                self.human_readable = true;
```

Comment:
Could this be an issue when a `HumanReadable<T>` is nested in a struct with non-wrapped types following it? e.g.

```
struct Test {
    human_readable: HumanReadable<Document>,
    non_human_readable: Document,
}
```

Reply:
Good catch!  I've fixed the logic and updated the test to flag that kind of failure.